### ISSUE TITLE
Keycloak ingress issue with static resources

### ISSUE THREAD

**Issue by rus666** [codecentric/helm-charts#133]
Hi,
I have an issue with following keycloak's ingress configuration:
```yaml
keycloak:
  basepath: "auth"
  baseprefix: "keycloak" #the ingress prefix value
  ...
  ingress:
    enabled: true
    path: /keycloak/?(.*)
  annotations:
     nginx.ingress.kubernetes.io/rewrite-target: /keycloak/$1
     nginx.ingress.kubernetes.io/ssl-redirect: "false"
     nginx.ingress.kubernetes.io/use-regex: "true"
```
There is a problem with reading static resources with that approach, the keycloak web app doesn't use configured ingress prefix. 
I added environment variable:
```yaml
env:
    - name: KEYCLOAK_CONTEXT_PATH
      value: {{ .Values.keycloak.baseprefix }}/{{ .Values.keycloak.basepath }}
```
according to that issue: https://issues.jboss.org/browse/KEYCLOAK-407, but unfortunatelyit didn't solve the problem. 
The web app used the correct url only after I changed the jboss config files inside the keycloak docker container and created the new docker image:
- /opt/jboss/keycloak/standalone/configuration/standalone.xml:
```xml
  <subsystem xmlns="urn:jboss:domain:keycloak-server:1.1">
          <web-context>keycloak/auth</web-context>
          ...
  </subsystem>
```
- /opt/jboss/keycloak/domain/configuration/domain.xml
```xml
  <subsystem xmlns="urn:jboss:domain:keycloak-server:1.1">
          <web-context>keycloak/auth</web-context>
          ...
  </subsystem>
```
Please, consider change jboss config files to fit the ingress rule.

Used version:
```yaml
name: keycloak
version: 6.0.0
appVersion: 7.0.0
```

**Comment by unguiculus** [comment]
The chart allows you to configure custom WildFly CLI scripts to achieve what you want.

**Comment by rus666** [comment]
```yaml
keycloak:
  cli:
    custom:
      {{ .Files.Get "scripts/custom.cli" }}
```
The script content:
`/subsystem=keycloak-server/:write-attribute(name=web-context,value=${env.KEYCLOAK_CONTEXT_PATH:keycloak/auth})` 
generates an item in `keycloak-startup` configmap, which seems to be OK.

The environment variable has correct value (`keycloak/auth`) inside the keycloak pod. However, the context path in jboss config files (/opt/jboss/keycloak/standalone/configuration/standalone.xml and /opt/jboss/keycloak/domain/configuration/domain.xml) is still `auth` instead of `keycloak/auth`.

I noticed that the `keycloak-startup` configmap values might be just ignored. I removed the `datasource.cli` from `values.yaml`
```yaml
datasource: |
      {{ .Files.Get "scripts/datasource.cli" }} - removed that line
```
datasource items were missing in `keycloak-startup` configmap and the keycloak launched without any warnings, because it used datasource values from environment variables.

Is there any other option to configure that web context path?

**Comment by rus666** [comment]
@unguiculus could you please comment on that?

**Comment by mbrdar** [comment]
```yaml
keycloak:
  basepath: "auth"

  ingress:
    enabled: true
    path: /keycloak/?(.*)
    hosts: ["your-host-name"]
  annotations:
    nginx.ingress.kubernetes.io/rewrite-target: /$1
    nginx.ingress.kubernetes.io/ssl-redirect: "false"
    nginx.ingress.kubernetes.io/use-regex: "true"

  cli:
    custom: |
      /subsystem=keycloak-server/:write-attribute(name=web-context,value=keycloak/auth)

  livenessProbe: |
    httpGet:
      path: keycloak/auth/
      port: http
    initialDelaySeconds: 120
    timeoutSeconds: 5

  readinessProbe: |
    httpGet:
      path: keycloak/auth/realms/master
      port: http
    initialDelaySeconds: 30
    timeoutSeconds: 1
```
This configuration is working for me.

**Comment by rus666** [comment]
values.yaml:
```yaml
cli:
    enabled: true
    nodeIdentifier: |
      {{ .Files.Get "scripts/node-identifier.cli" }}
    logging: |
      {{ .Files.Get "scripts/logging.cli" }}
    ha: |
      {{ .Files.Get "scripts/ha.cli" }}
    datasource: |
      {{ .Files.Get "scripts/datasource.cli" }}
    # Custom CLI script
    custom: |
     {{ .Files.Get "scripts/custom.cli" }}
```
custom.cli:
`/subsystem=keycloak-server/:write-attribute(name=web-context,value=keycloak/auth)`

Unfortunately one of `custom.cli`'s preceding scripts had some syntax error and other scripts following the invalid one didn't execute (without any warning/error). I removed all apart from the `custom.cli` and it worked just as in your example.

I think that it would be really nice to have some error logging instead of just swallowing them.
